src/AuthContext.tsx

```diff
@@ -104,6 +104,18 @@ export const AuthProvider: FC<AuthProviderProps> = ({
     getUser();
   }, [location]);
 
+  useEffect(() => {
+    // for refreshing react state when new state is available in e.g. session storage
+    const updateUserData = async () => {
+      const user = await userManager.getUser();
+      setUserData(user);
+    }
+
+    userManager.events.addUserLoaded(updateUserData);
+
+    return () => userManager.events.removeUserLoaded(updateUserData);
+  }, [userManager])
```

Comment:
@Powersource  Could you explain please why would userManager change if it has an execution path where its ignoring props? line #46

Reply:
I'm no react pro so not gonna defend this too much. I believe https://www.npmjs.com/package/eslint-plugin-react-hooks told me to change that line to this.

But the user could switch out the usermanager they're passing in as a prop, line 36. And then we'd need to re-run this Effect to get it to apply the event handler on the right usermanager.